nuclear/b.py

```diff
@@ -85,50 +85,69 @@
     )
     subcommands.required = True
 
-    # Get all of the packages that are in the build tools
-    modules = list(pkgutil.iter_modules(path=[nuclear_tools_path, user_tools_path]))
-
-    # First we try to see if sys.argv[1] gives us all the information we need
-    # If it does we only need to load that module directly
-    # Otherwise we load every module so we can build a help for possible tools
-    target_modules = [] if len(sys.argv) < 2 else [m for m in modules if not m[2] and m[1] == sys.argv[1]]
-    modules = target_modules if len(target_modules) > 0 else modules
-
-    # Our tools dictionary
+    # Look through the various tools to see if we can find one that matches our arguments
+    # If we do we don't need to load all the tools and can just trigger this one directly
+    # This saves importing things we don't need
+    for path in [user_tools_path, nuclear_tools_path]:
+        for dirpath, dnames, fnames in os.walk(path):
+            for f in fnames:
+                if f != "__init__.py" and f.endswith(".py"):
+
+                    # Check if this is the tool for the job
+                    components = os.path.relpath(os.path.join(dirpath, f[:-3]), path).split(os.sep)
+                    if sys.argv[1 : len(components) + 1] == components:
+
+                        # Load the module
+                        fqn = ".".join(components)
```

Comment:
Whats `fqn` for?

Reply:
It's for the next few lines where I instead of using fqn just use the same thing again!